SignalExtraction/fitWm.C

```diff
@@ -744,167 +746,205 @@ void fitWm(const TString  outputDir,   // output directory
-              hWmunuMet->Fill(corrMet,weight);
-              hWmunuMet_PileupUp->Fill(corrMet,weightUp);
-              hWmunuMet_PileupDown->Fill(corrMet,weightDown);
-              if(q>0) {
-                recoilCorr->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lepPt,lep->Phi(),pU1,pU2,0);
-                hWmunuMetp->Fill(corrMet,weight); corrMet=met, corrMetPhi=metPhi;
-                recoilCorrUp->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lepPt,lep->Phi(),pU1,pU2,0);
-                hWmunuMetp_PileupUp->Fill(corrMet,weightUp);corrMet=met, corrMetPhi=metPhi;
-                recoilCorrDown->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lepPt,lep->Phi(),pU1,pU2,0);
-                hWmunuMetp_PileupDown->Fill(corrMet,weightDown);
-                corrMet=met, corrMetPhi=metPhi;
-              } else { 
-                recoilCorrm->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lepPt,lep->Phi(),pU1,pU2,0);
-                hWmunuMetm->Fill(corrMet,weight); corrMet=met, corrMetPhi=metPhi;
-                recoilCorrmUp->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lepPt,lep->Phi(),pU1,pU2,0);
-                hWmunuMetm_PileupUp->Fill(corrMet,weightUp);corrMet=met, corrMetPhi=metPhi;
-                recoilCorrmDown->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lepPt,lep->Phi(),pU1,pU2,0);
-                hWmunuMetm_PileupDown->Fill(corrMet,weightDown);corrMet=met, corrMetPhi=metPhi;
-                corrMet=met, corrMetPhi=metPhi;
-              }
-              corrMet=met, corrMetPhi=metPhi;
-            }
-//             recoilCorr.Correct(corrMet,corrMetPhi,genVPt,genVPhi,lepPt,lep->Phi(),1,q);
-            hWmunuMet_RecoilUp->Fill(corrMet,weight);
-            if(q>0) {
-              pU1 = 0; pU2 = 0; 
-              hWmunuMetp_RecoilUp->Fill(corrMet,weight); 
-              corrMet=met, corrMetPhi=metPhi;
-            } else { 
-              pU1 = 0; pU2 = 0; 
-              hWmunuMetm_RecoilUp->Fill(corrMet,weight);
-              corrMet=met, corrMetPhi=metPhi;
-            }
-            hWmunuMet_RecoilDown->Fill(corrMet,weight);
+            corrMet=met, corrMetPhi=metPhi;
+            hWmunuMet->Fill(corrMet,weight);
+            hWmunuMet_PileupUp->Fill(corrMet,weightUp);
+            hWmunuMet_PileupDown->Fill(corrMet,weightDown);
             if(q>0) {
-              pU1 = 0; pU2 = 0; 
-              hWmunuMetp_RecoilDown->Fill(corrMet,weight);
+              //recoilCorr->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lepPt,lep->Phi(),pU1,pU2,0);
+              if(fabs(genVy)<0.5)
+                recoilCorr05->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lepPt,lep->Phi(),pU1,pU2,0);
+              else if (fabs(genVy)>=0.5 && fabs(genVy)<1.0)
+                recoilCorr051->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lepPt,lep->Phi(),pU1,pU2,0);
+              else
+                recoilCorr1->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lepPt,lep->Phi(),pU1,pU2,0); 
```

Comment:
lepPt --> should be something like vLepRaw->Pt()

This need to be done everyWhere